File: Graph Theory/BFS.py
def traversal(s):
    idx = 0
    queue = []
    queue.append(s)
    dist[source] = 0
    while idx < len(queue): # equivalent to while(!q.empty())
        u = queue[idx]      # equivalent to queue.front()
        idx += 1            # equivalent to pop()
        for v in adj[u]:
            if dist[v] == -1:
                dist[v] = dist[u] + 1
                queue.append(v)

V, E = [int(x) for x in input().split()]
dist = [-1 for _ in range(V+1)]
adj = [[] for _ in range(V+1)]
for _ in range(E):
    u, v, w = [int(x) for x in input().split()] # weight is not used for BFS
    adj[u].append(v)
    adj[v].append(u)

traversal(1)

for i in range(1,n+1):
    print(f"dist of {i} = {dist[i]}")

File: Graph Theory/BFS.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <queue> // for queue and priority queue
using namespace std;

int V, E;
vector<vector<int>> adj;
vector<int> dist;

void traversal(int s) {
	dist[s] = 0;
	queue<int> q;
	q.push(s);
	while (!q.empty()) {
		int u = q.front();
		q.pop();
		for (int v : adj[u]) {
			if (dist[v] == -1) {
				dist[v] = dist[u] + 1; // for unweighted graph
				q.push(v);
			}
		}
	}
}

int main() {
	cin >> V >> E;
	adj.resize(V+1);
	dist.resize(V+1);
	for (int i = 1; i <= V; i++)
		dist[i] = -1;
	for (int i = 0; i < E; i++) {
		int u, v, w;
		cin >> u >> v >> w; // weight is not used for BFS
		adj[u].push_back(v);
		adj[v].push_back(u);
	}
	traversal(1);
	for (int i = 1; i <= V; i++)
		cout << "dist of " << i << " = " << dist[i] << endl;
}
